services/datalad/datalad_service/tasks/publish.py

```diff
@@ -168,22 +168,40 @@ def export_all_tags(store, dataset, cookies, get_realm, check_should_export, esL
         clear_dataset_cache(dataset, cookies)
 
 def check_remote_has_version(dataset, remote, tag):
-    response = dataset.repo._run_annex_command(
-        'info',
-        annex_options=[
-            '--json',
-            tag,
-        ]
-    )
     try:
-        
+        # extract remote uuid from `git annex info <tag>`
+        response = dataset.repo._run_annex_command(
+            'info',
+            annex_options=[
+                '--json',
+                tag,
+            ]
+        )
         info = json.loads(response[0])
         remotes = info.get('repositories containing these files', [])
         s3_PUBLIC_remote = [r for r in remotes if r.get('description') == f'[{remote}]']
-        return bool(s3_PUBLIC_remote)
-    except AttributeError as err:
-        logger.error(err)
+        remote_id_A = s3_PUBLIC_remote and s3_PUBLIC_remote[0].get('uuid')
```

Comment:
This isn't necessarily the public remote right?

Reply:
changed to `remote_repo`